### ISSUE TITLE
Flaky-test: TxnLogBufferedWriterTest.testMainProcess

### ISSUE THREAD

**Issue by codelipenghui** [apache/pulsar#16989]
[example failure](https://github.com/apache/pulsar/runs/7720437968?check_suite_focus=true)

```
  Error:  Tests run: 25, Failures: 2, Errors: 0, Skipped: 13, Time elapsed: 21.138 s <<< FAILURE! - in org.apache.pulsar.transaction.coordinator.impl.TxnLogBufferedWriterTest
  Error:  testMainProcess(org.apache.pulsar.transaction.coordinator.impl.TxnLogBufferedWriterTest)  Time elapsed: 0.12 s  <<< FAILURE!
  java.lang.AssertionError: expected [1349] but found [1512]
  	at org.testng.Assert.fail(Assert.java:99)
  	at org.testng.Assert.failNotEquals(Assert.java:1037)
  	at org.testng.Assert.assertEqualsImpl(Assert.java:140)
  	at org.testng.Assert.assertEquals(Assert.java:122)
  	at org.testng.Assert.assertEquals(Assert.java:907)
  	at org.testng.Assert.assertEquals(Assert.java:917)
  	at org.apache.pulsar.transaction.coordinator.impl.TxnLogBufferedWriterTest.testMainProcess(TxnLogBufferedWriterTest.java:242)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
  	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
  	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:45)
  	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
  	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
  	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
  	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
  	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
  	at java.base/java.lang.Thread.run(Thread.java:833)
```

<!-- optionally provide the full stacktrace ->
<details>
<summary>Full exception stacktrace</summary>
<code><pre>
full exception stacktrace here
</pre></code>
</details>

**Comment by codelipenghui** [comment]
Not the same issue with https://github.com/apache/pulsar/issues/16723

**Comment by poorbarcode** [comment]
assigned to me

**Comment by mattisonchao** [comment]
I think this problem may caused by #16968 and fixed by #17018.
Currently, the fix PR already merged, maybe we can close this issue until we got it again.
